### DISCUSSION TITLE
foxglove stuidio support multiple mcap file playing

### DISCUSSION THREAD

**Discussion by kyoiora** [foxglove/community#865] [Q&A] [answered]
Now, Foxglove only support selecting one mcap file to play  in one time. But in our scenario ,  multiple socs are used to recording data at the same time.  And serveral mcap files are generated  with synced timesamp.

What we are doing is to have a backend service to dynamicly merging data into one big mcap file for vizualization， which will generate lots of temp files and speed of opening file is slow due to the process of data.

Question: any plan for foxglove studio supporting multi mcap files playing?

**Comment by mkerpelman** [comment, marked as answer]
Hello, as of now Foxglove Studio continues to allow visualization one mcap file at a time. However, our [Team](https://foxglove.dev/pricing) plan allows you to store those multiple mcap files in a hosted cloud location that interfaces with Foxglove Studio. When you choose to visualize data for a particular device or asset, all of the data associated with that the device for the time window of your choosing is made available for visualization at the same time, regardless of which mcap file that data originally lived in.

**Reply by mrkbac** [reply]
Are there any plans to support this without using the online offering? Uploading multiple mcaps only to download them again while viewing is a sub-optimal workflow, especially for one-off mcaps and/or on bad internet connections.

**Reply by defunctzombie** [reply]
There are no short-term plans for this - as a workaround you can use the `mcap merge` command (from the [CLI](https://mcap.dev/guides/cli)) to merge your files into one and open that file.
